src/main/scala/devsearch/features/FeatureRecognizer.scala

```diff
@@ -87,7 +87,13 @@ object FeatureRecognizer extends (CodeFile => TraversableOnce[Feature]) with jav
     ValDefExtractor
   )
 
-  def apply(data: CodeFile) = extractors.flatMap(_.extract(data))
+  def apply(data: CodeFile): Set[Feature] = {
+    extractors.flatMap { case extractor =>
+      scala.util.Try {
```

Comment:
I know more or less but there are quite a few different ones. They can be found by "grep throw" in the normalized package :) It would probably make more sense to place the try-catch in the SemanticExtractor to avoid catching stuff we don't want to. I was actually thinking of injecting the normalization into the CodeFile as a lazy field alongside the AST but this would be for future work.

However, these are just details, I find this version fine (especially so close to code freeze).